File: app/src/main/java/com/manimarank/spell4wiki/databases/dao/WordsHaveAudioDao.java
package com.manimarank.spell4wiki.databases.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.manimarank.spell4wiki.databases.entities.WordsHaveAudio;

import java.util.List;

@Dao
public interface WordsHaveAudioDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(WordsHaveAudio... wordsHaveAudios);

    @Update
    public void update(WordsHaveAudio... wordsHaveAudios);

    @Delete
    public void delete(WordsHaveAudio wordsHaveAudio);

    @Query("SELECT word FROM words_already_have_audio WHERE language_code = :code")
    public List<String> getWordsAlreadyHaveAudioByLanguage(String code);

}

File: app/src/main/java/com/manimarank/spell4wiki/data/db/dao/WordsHaveAudioDao.kt
package com.manimarank.spell4wiki.data.db.dao

import androidx.room.*
import com.manimarank.spell4wiki.data.db.entities.WordsHaveAudio

@Dao
interface WordsHaveAudioDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(vararg wordsHaveAudios: WordsHaveAudio?)

    @Update
    fun update(vararg wordsHaveAudios: WordsHaveAudio?)

    @Delete
    fun delete(wordsHaveAudio: WordsHaveAudio?)

    @Query("SELECT word FROM words_already_have_audio WHERE language_code = :code")
    fun getWordsAlreadyHaveAudioByLanguage(code: String?): ArrayList<String>?
}